.github/scripts/shellCheck.sh

```diff
@@ -1,48 +1,49 @@
 #!/bin/bash
 
-CURRENT_DIR=$(pwd)
-ROOT_DIR="$(dirname "$(dirname "$(dirname "$(realpath "${BASH_SOURCE[0]}")")")")"
-
-cd "$ROOT_DIR" || exit 1
-
-source ./.github/scripts/shellUtils.sh
-
-declare -r DIRECTORIES_TO_IGNORE=(
-  './node_modules'
-  './vendor'
-  './ios/Pods'
-  './.husky'
+SCRIPT_DIR=$(cd -- "$(dirname -- "${BASH_SOURCE[0]}")" &> /dev/null && pwd)
+readonly SCRIPT_DIR
+ROOT_DIR="$(dirname "$(dirname "$SCRIPT_DIR")")"
+readonly ROOT_DIR
+
+source "$SCRIPT_DIR"/shellUtils.sh
+
+readonly DIRECTORIES_TO_IGNORE=(
+    "$ROOT_DIR/node_modules"
+    "$ROOT_DIR/vendor"
+    "$ROOT_DIR/ios/Pods"
+    "$ROOT_DIR/.husky"
 )
 
 # This lists all shell scripts in this repo except those in directories we want to ignore
 read -ra IGNORE_DIRS < <(join_by_string ' -o -path ' "${DIRECTORIES_TO_IGNORE[@]}")
-SHELL_SCRIPTS=$(find . -type d \( -path "${IGNORE_DIRS[@]}" \) -prune -o -name '*.sh' -print)
-info "👀 Linting the following shell scripts using ShellCheck: $SHELL_SCRIPTS"
-info
+SHELL_SCRIPTS=$(find "$ROOT_DIR" -type d \( -path "${IGNORE_DIRS[@]}" \) -prune -o -name '*.sh' -print)
+info "👀 Linting the following shell scripts using ShellCheck:"
+echo "$SHELL_SCRIPTS"
+echo
 
-ASYNC_PROCESSES=()
+PIDS=()
 for SHELL_SCRIPT in $SHELL_SCRIPTS; do
-  if [[ "$CI" == 'true' ]]; then
-    # ShellCheck is installed by default on GitHub Actions ubuntu runners
-    shellcheck -e SC1091 "$SHELL_SCRIPT" &
-  else
-    # Otherwise shellcheck is used via npx
-    npx shellcheck -e SC1091 "$SHELL_SCRIPT" &
-  fi
-  ASYNC_PROCESSES+=($!)
+    if [[ "$CI" == 'true' ]]; then
+        # ShellCheck is installed by default on GitHub Actions ubuntu runners
+        shellcheck -e SC1091 "$SHELL_SCRIPT" &
+    else
+        # Otherwise shellcheck is used via npx
+        npx shellcheck -e SC1091 "$SHELL_SCRIPT" &
+    fi
+    PIDS+=($!)
 done
 
 EXIT_CODE=0
-for PID in "${ASYNC_PROCESSES[@]}"; do
+for PID in "${PIDS[@]}"; do
   if ! wait "$PID"; then
     EXIT_CODE=1
   fi
 done
 
-cd "$CURRENT_DIR" || exit 1
-
-if [ $EXIT_CODE == 0 ]; then
-  success "ShellCheck passed for all files!"
+if [[ $EXIT_CODE == 0 ]]; then
+    success 'ShellCheck passed for all files!'
+else
+    error 'ShellCheck failed for one or more files'
 fi
 
 exit $EXIT_CODE
```

Comment:
Similar to above, you can simplify a bit by inverting the logic. Also it's more idiomatic to use `-ne`, `-eq`, `-gt`, etc for numeric comparisons and reserve "==" for string comparisons
```suggestion
if [[ $EXIT_CODE -ne 0 ]]; then
    error 'ShellCheck failed for one or more files'
    exit $EXIT_CODE
fi

success 'ShellCheck passed for all files!'
```

Reply:
done